lib/public/components/detector/DetectorSelectionDropdownModel.js

```diff
@@ -31,13 +31,14 @@ export class DetectorSelectionDropdownModel extends SelectionDropdownModel {
      * @inheritDoc
      */
     _initialize() {
-        detectorsProvider.getAll().then(
-            (detectors) => {
-                this.availableOptions = RemoteData.success(detectors.map(({ name }) => ({ value: name })));
-            },
-            (errors) => {
-                this.availableOptions = RemoteData.failure(errors);
-            },
-        );
+        this._detectorsProvider = createDetectorsProviderInstance();
+        this._detectorsProvider.bubbleTo(this);
+        this._detectorsProvider.fetch()
```

Comment:
The whole purpose of the provider was to hide the `fetch` stuff: A provider is simply a way to get a remote data.
(Now that we have more tooling, it may be replaced by a datasource, TO DISCUSS)

Reply:
1. I need source of detector for active columns
2. Previously, getAll() did fetch
3. Indeed, in case of this provider, there is no reason to fetch detectors each time they are needed, so they are stored, but it's good to have mechanism of notifying front about the state of request, so it's observable, but I think what's not good is to have single instance of detectorsProvider which notifies many components at once, therefore there is factory for detectors providers which shares single RemoteData<Detector[]> (parentInstance) (renamed)